client/me/purchases/purchases-list-in-dataviews/index.tsx

```diff
@@ -0,0 +1,206 @@
+import PurchasesSite from '../purchases-site';
+import { purchasesDataFields } from './purchases-data-field';
+import { purchasesDataView } from './purchases-data-view'; // This is the temporary data
+import PurchasesListHeader from './purchases-list-header';
+import './style.scss';
+
+const useDataViewPurchasesList = config.isEnabled( 'purchases/purchase-list-dataview' );
+
+export interface PurchasesListProps {
+	noticeType?: string | undefined;
+}
+
+export interface PurchasesListConnectedProps {
+	hasLoadedUserPurchasesFromServer: boolean;
+	isFetchingUserPurchases: boolean;
+	purchases: Purchase[] | null;
+	subscriptions: MembershipSubscription[];
+	sites: SiteDetails[];
+	nextAppointment: NextAppointment | null;
+	isUserBlocked: boolean;
+	availableSessions: number[];
+	siteId: number | null;
+}
+
+class PurchasesListDataView extends Component<
+	PurchasesListProps & PurchasesListConnectedProps & WithStoredPaymentMethodsProps & LocalizeProps
+> {
+	isDataLoading() {
+		if ( this.props.isFetchingUserPurchases && ! this.props.hasLoadedUserPurchasesFromServer ) {
+			return true;
+		}
+
+		return ! this.props.sites.length && ! this.props.subscriptions.length;
+	}
+
+	renderConciergeBanner() {
+		const { nextAppointment, availableSessions, isUserBlocked } = this.props;
+		return (
+			<PurchaseListConciergeBanner
+				nextAppointment={ nextAppointment ?? undefined }
+				availableSessions={ availableSessions }
+				isUserBlocked={ isUserBlocked }
+			/>
+		);
+	}
+
+	renderMembershipSubscriptions() {
+		const { subscriptions } = this.props;
+
+		if ( ! subscriptions.length || this.isDataLoading() ) {
+			return null;
+		}
+
+		return getSubscriptionsBySite( subscriptions ).map( ( site ) => (
+			<MembershipSite site={ site } key={ site.id } />
+		) );
+	}
+
+	render() {
+		const { purchases, sites, translate, subscriptions } = this.props;
+		let content;
+
+		if ( this.isDataLoading() ) {
+			content = <PurchasesSite isPlaceholder />;
+		}
+
+		const onChangeView = () => {
+			alert( 'You clicked something!!' );
+		};
+
+		const getItemId = ( item: Purchases.Purchase ) => {
+			return item.id.toString();
+		};
+
+		if ( purchases && purchases.length ) {
+			if ( useDataViewPurchasesList ) {
+				content = (
+					<Card id="purchases-list" className="section-content" tagName="section">
+						<DataViews
+							data={ purchases }
+							fields={ purchasesDataFields }
+							view={ purchasesDataView }
+							onChangeView={ onChangeView }
+							defaultLayouts={ { table: {} } }
+							actions={ undefined }
+							getItemId={ getItemId }
+							paginationInfo={ { totalItems: 100, totalPages: 10 } }
+						/>
+					</Card>
+				);
+			} else {
+				content = (
+					<>
+						{ this.renderConciergeBanner() }
+
+						<PurchasesListHeader showSite />
+
+						{ getPurchasesBySite( purchases, sites ).map( ( site ) => (
+							<PurchasesSite
+								key={ site.id }
+								siteId={ site.id }
+								name={ site.name }
```

Comment:
We removed the `name` prop in https://github.com/Automattic/wp-calypso/pull/101537 because it turns out it wasn't being used.

Reply:
FTR here's where it was "used" previously

https://github.com/Automattic/wp-calypso/pull/101537/files#diff-c84f5a076b184965f0e310e29d81f7359b682c2259a08f4847a621d8988338daL143-L149